pkg/cluster/internal/providers/podman/provision.go

```diff
@@ -252,12 +252,23 @@ func getProxyEnv(cfg *config.Cluster, networkName string) (map[string]string, er
 			return nil, err
 		}
 		noProxyList := append(subnets, envs[common.NOProxy])
-		// Add pod and service dns names to no_proxy to allow in cluster
+		// Add pod,service and all the cluster nodes' dns names to no_proxy to allow in cluster
 		// Note: this is best effort based on the default CoreDNS spec
 		// https://github.com/kubernetes/dns/blob/master/docs/specification.md
 		// Any user created pod/service hostnames, namespaces, custom DNS services
 		// are expected to be no-proxied by the user explicitly.
+		var clusterNodeNames []string
```

Comment:
we should probably just restructure to pass these in or add them to the list at the call site?

Reply:
I was thinking in adding them here

https://github.com/kubernetes-sigs/kind/blob/b6bc112522651d98c81823df56b7afa511459a3b/pkg/cluster/internal/providers/common/proxy.go#L41-L63

that will cascade for all the providers

Reply:
The docker implementation already has this AFAICT, we just need to modify podman to match and drop the docker changes, I think https://github.com/kubernetes-sigs/kind/pull/2885#discussion_r950321305

Reply:
If we were to match the podman flow with the docker one in that regard I guess the best option to start the restructuring from the `planCreation` and trickle it down. I'd be happy to do that.

@BenTheElder  @aojea : are you fine with that direction?

Reply:
I've just pushed a fresh commit with the refactoring. In short - it's an alignment with what we have for docker in that regards. I  must admit I like this version a lot more than the previous one. I hope you don't disagree ;-)

/cc @BenTheElder @aojea